executable/elf.ksy

```diff
@@ -700,6 +703,258 @@ types:
+              unregistered private use.
+          - id: vendor_data
+            size-eos: true
+            if: not is_public
+          - id: public_subsection
+            size-eos: true
+            type: arm_attributes_public_subsection
+            if: is_public
+        instances:
+          is_public:
+            value: vendor_name == "aeabi"
+        types:
+          arm_attributes_public_subsection:
+            seq:
+              - id: type
+                type: u1
+                enum: types
+                doc: |
+                  It is not clear from the Arm ABI whether this should be a
+                  single byte or a ULEB128 integer, but it's more likely that it
+                  is meant to be one byte, see the following references:
+
+                  * <https://developer.arm.com/documentation/ihi0045/h?lang=en>\
+                    "ADDENDUM: Build Attributes > Representing build attributes
+                    in ELF files > **Formal syntax of a
+                    public ("aeabi") attributes subsection**":
+
+                    > Byte-size includes the initial tag **byte**, the size
+                    > field itself, [...]
+
+                  * <https://github.com/llvm/llvm-project/blob/014c6b07362c/llvm/lib/Support/ELFAttributeParser.cpp#L139>
+              - id: len_attributes
+                type: u4
+              - id: attributes
+                size: len_attributes - type._sizeof - len_attributes._sizeof
+                type: arm_attributes_wrapper
+            enums:
+              types:
+                1: file
+                2: section
+                3: symbol
+          arm_attributes_wrapper:
+            seq:
+              - id: attributes
+                type: arm_attribute
+                repeat: eos
+          arm_attribute:
+            -webide-representation: '{tag} - s:{value_str} i:{value_int}'
+            seq:
+              - id: tag_raw
+                type: vlq_base128_le
+              - id: value_str
+                type: strz
+                encoding: ASCII
+                if: is_value_str
+              - id: value_int_raw
+                type: vlq_base128_le
+                if: is_value_int
+            instances:
+              value_int:
+                value: value_int_raw.value
+                if: is_value_int
+              tag:
+                value: tag_raw.value
+                enum: tags
+              is_value_str:
+                value: |
+                  tag.to_i < 32 ? (tag == tags::cpu_raw_name or tag == tags::cpu_name)
+                  : tag.to_i % 2 != 0
+                doc-ref: https://github.com/llvm/llvm-project/blob/014c6b07362c/lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.cpp#L1260-L1276
+              is_value_int:
+                value: not is_value_str
+            enums:
+              # https://github.com/llvm/llvm-project/blob/014c6b07362c/llvm/include/llvm/Support/ARMBuildAttributes.h#L34-L85
+              tags:
```

Comment:
How about putting large enums into own ksy files?

Reply:
> How about putting large enums into own ksy files?

This would be nice sometime in the future, but I'm afraid our infrastructure isn't ready for that yet. There are problems in several languages with trying to access an enum in an imported spec, see test `EnumImport` in the [CI dashboard](https://ci.kaitai.io/). And even when this is fixed, we should figure out what to do with these import-only specs at https://formats.kaitai.io/ - they should be at least marked with some special attribute that will exclude them from the normal view. There is no point in displaying such specifications on homepage, that would be confusing.

So for now, let's maintain the status quo.

Reply:
>they should be at least marked with some special attribute that will exclude them from the normal view.

I guess it is trivial to detect them automatically.